### ISSUE TITLE
valid-jsdoc seems to fail when using @external and @function

### ISSUE THREAD

**Issue by cyChop** [eslint/eslint#6694]
**What version are you using?** ESLint 3.0.1

**What did you do?**

``` javascript
    /**
     * The jQuery plugin namespace.
     *
     * @external "jQuery.fn"
     * @see {@link http://learn.jquery.com/plugins/|jQuery Plugins}
     */

    // This is a problem...
    // eslint-disable-next-line valid-jsdoc
    /**
     * Creates a {@link BeveragesView} in the selected elements.
     *
     * @function external:"jQuery.fn".beverages
     * @param {Object} options the options for the beverages view
     */
```

**What happened?**
I had to add `eslint-disable`.

Removing the line beginning with `@function` removes the error. Trying synonyms (`@func` and `@method`) seems to have no impact. Moving the line around does not fix the error.

I don't have more information on what makes this JSDoc invalid, so this may be normal but I don't know what's wrong.

**What did you expect to happen?**
This should be valid (I think; see [JSDoc's doc](http://usejsdoc.org/tags-external.html)).

**Comment by platinumazure** [comment]
Marking as a question for now. Hopefully someone who knows JSDoc on the team will take a look and speak to whether this might be a bug or not.

**Comment by nzakas** [comment]
@cyChop can you please provide the actual output from ESLint that you're seeing? (I'm assuming you're reporting what you think is improper ESLint output since your expectation is that it's valid.)

**Comment by cyChop** [comment]
@nzakas Yes, I basically copied an example from the JSDoc's documentation, so I expected it to be valid.

Here is what ESLint tells me though:

```
X:\some\path\beverages.js
  21:5  error  JSDoc syntax error  valid-jsdoc
```

As said above, I have no more information, so I don't know at all what I can do to fix the error.

Same error reported using gulp-eslint or eslint as integrated within IntelliJ IDEA (which is logical since they both use the installed ESLint module).

**Comment by nzakas** [comment]
My hunch is that the parser can't parse `external:"jQuery.fn".beverages`. Do you need the quotes around jQuery.fn?

**Comment by cyChop** [comment]
Your hunch appears to be correct. Removing the quotes does suppress the error.

But it also makes my function disappear from the generated doc, so I guess I do need them.

**Comment by nzakas** [comment]
Okay, well this is a problem in Doctrine (the JSDoc parser) rather than in ESLint, so there's not much we can do on our side alone. Would you mind opening a bug on the Doctrine repo? https://github.com/eslint/doctrine/issues

We can leave this issue open just in case any other ESLint user ends up with the same problem.

**Comment by cyChop** [comment]
Done. [Direct link to the Doctrine issue](https://github.com/eslint/doctrine/issues/166).

Thanks for having taken the time to look in on that.

**Comment by Hibou57** [comment]
Note the same as the OP, but related I believe, and may not be worth a separate issue.

Trying this:

    /**
     * @external document
     */
    
    /** Foo
     * @return {HTMLElement} Created element
     */
    function foo() {
       return document.createElement("SPAN");
    }

I get:

> 'document' is not defined. [Error/no-undef]

I though it would understand `@external` and it does not. It’s OK if I add `/*global document */`, which is just redundant.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
